_posts/2017-07-28-reproducible-research-badges.md

```diff
@@ -0,0 +1,154 @@
+
+![extender config](/public/images/2017-07-28-badges/extender_configuration.png "Figure 5: o2r-extender configuration")
+<p class="attributionInlineImage">Figure 5: o2r-extender configuration</p>
+
+
+## Outlook: Action integrations
+
+The extender also has a feature that has nothing to do with badges at all. In the context of open science and reproducible research, we have place the reproducibility service in a larger context as described in the [o2r architecture](http://o2r.info/architecture/) (see section Business context). Two core aspects are loading research workspaces from cloud storage and connecting to suitable data repositories for actual storage of ERCs.
+
+To facilitate this integration for users, the extender can also augment the user interfaces  of the cloud collaboration platform [Sciebo](http://sciebo.de/) and the scientific data repository [Zenodo](https://zenodo.org/) to integrate our reproducibility service.
+
+When using Sciebo, a button is added  to a file’s or directory’s context menu. It allows direct submission of workspaces to the o2r platform to create a new Executable Research Compendium (ERC): 
+
+![sciebo integration](/public/images/2017-07-28-badges/sciebo_integration.png "Figure 6: Sciebo upload integration")
+<p class="attributionInlineImage">Figure 6: Sciebo upload integration</p>
+
+On the other hand, when you are viewing an Executable Research Compendium on Zenodo, a small badge redirects to the inspection view in the o2r platform:
+
+![zenodo integration](/public/images/2017-07-28-badges/zenodo_integration.png "Figure 7: Zenodo inspection integration")
+<p class="attributionInlineImage">Figure 7: Zenodo inspection integration</p>
+
+## Challenges
+
+The study project [Badges for computational geoscience containers](https://zivgitlab.uni-muenster.de/geocontainer-badges) initially implemented eight microservices responsible for the six different badges types, badge scaling and testing. A microservice architecture using Docker containers was not chosen because the project needs immense scaling capabilities, but for another reason: developing independent microservices makes work organization much easier, especially for a study project where students prefer different programming languages and have different skillsets. However, for the o2r project, the eight microservices needed to become a single microservice. This required refactoring, rewriting and bug fixing. Now, when a badge is requested, a [promise chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) is called depending on the badge type. It consists of a number of functions of which some are used by all badges. The functions each contain code written in the study project separated into small chunks to avoid a [callback hell](http://callbackhell.com/).
```

Comment:
Add link to source code when talking about the code (e.g. on the promise chain description).